opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -388,13 +384,22 @@ fn resolve_http_endpoint(
         return Ok(endpoint);
     }
 
+    let provided_endpoint_clone = provided_endpoint.clone();
```

Comment:
Is the clone here necessary?
If it is, could it be moved into the map_err body so its only done when it's needed?